### PR TITLE
Markdown Rendering Feature

### PR DESCRIPTION
**Pull request by kfrankc**
This PR allows StaticHandler to handle markdown files to be rendered to HTML when served on the webserver.

### CODE REVIEW

File: request_handlers.cc [hunk 1]
```diff
@@ -129,7 +130,21 @@ RequestHandler::Status StaticRequestHandler::handleRequest(const Request& reques
...
-  response->SetBody(reply);
+
+  if (extension == "md") {
+    markdown::Document doc;
+    doc.read(reply);
+
+    // markdown::write write out to an ostream, so we need to convert it to a string
+    std::ostringstream stream;
+    doc.write(stream);
+    std::string md_html =  stream.str();
```
Comment (KyCodeHuynh) [inline request_handlers.cc]: Very minor, but it looks like there's an extra space after the equals-sign.

File: test/test.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+1. Numbered
+  * sub-list
+2. List
+3. Works
+
+---
+
+    Code through indent
+
+___
```
Comment (KyCodeHuynh) [inline test/test.md]: It looks like there was some macOS "Smart Substitution" here, as the `---` has been replaced by `——`.

File: request_handlers.cc [hunk 3]
```diff
@@ -129,7 +130,21 @@ RequestHandler::Status StaticRequestHandler::handleRequest(const Request& reques
...
+
+    response->SetBody(md_html);
+  }
+  else {
+    response->SetBody(reply);
```
Comment (KyCodeHuynh) [inline request_handlers.cc]: What do you think of switching this if-else order? I personally find it easier to get lost among the Markdown handling code if it happens first. 

Then we'd have: 

```c++
if (!md) {
  // Non-Markdown case
} else {
  // Markdown-handling
}
```

Comment (apfong) [inline request_handlers.cc]: Yeah we can do that, in this case reading it either way is the same to me.

File: test/test.md (hunk 2 above)
Comment (apfong) [inline test/test.md]: Ah yeah its a little confusing with github's font style. Markdown supports using either three `*`, `-`, or `_` and this is the last case.

Comment (kfrankc) [inline test/test.md]: interesting; never realized macOS has these smart substitutions in text editors too haha

Comment (KyCodeHuynh) [inline test/test.md]: Ah, so these are actually underscores?

**Pull request merged**